### ISSUE TITLE
feat: automatically inject a CA trust bundle into common pod locations

### ISSUE THREAD

**Issue by Racer159** [defenseunicorns/uds-core#465]
### Is your feature request related to a problem? Please describe.

**As** Kay **I want** a CA bundle injected into my application's pod **so that** (in most cases) I do not need to worry about wiring this up manually in Kubernetes manifests / Helm charts.

### Describe the solution you'd like

- **Given** I have an application I am developing that is build on `uds-core`
- **When** I deploy the application (without manually wiring up https://github.com/defenseunicorns/uds-core/issues/464)
- **Then** Pepr injects the certificate bundle in the namespace into my pod containers in common places for applications (`debian`/`rhel`/`alpine`/`java`)

### Describe alternatives you've considered

We could not do this and instead have documentation on how to wire this up in `uds-core` manually and provide that as part of YAYE.

### Additional context

This is related / blocked on https://github.com/defenseunicorns/uds-core/issues/464

**Comment by blancharda** [comment]
Here's a [very official accounting](https://serverfault.com/a/722646) of a bunch of places it could be mounted..

We could blast mount points at every folder we think we might encounter, but that feels messy.. 
Maybe as part of the package definition a maintainer could specify which mount points and/or formats to use. If we really wanted to get fancy, it could include a pod selector as well so that individual pods could be handled differently.. but I'm not sure how much it matters if the CA bundle is duplicated into multiple (unused) folders.

**Comment by mjnagel** [comment]
I'm not entirely opposed to doing this, but as @blancharda mentioned there are a number of spots it could be mounted and I think we'd need to be careful to not wipe out other potential trusted CAs that are built into the images?

Regardless I think https://github.com/defenseunicorns/uds-core/issues/464 is probably a better first target as mentioned and we can see if that satisfies most downstream needs. I would hope that we could standardize some mutation but definitely would need more care as that's a pretty widespread one.

**Comment by blancharda** [comment]
> wipe[ing] out other potential trusted CAs that are built into the images

That might be a feature tbh.. 🤔 
It would mean ALL trusted sources would have to come from the trust-manager configuration.

But that aside.. I would hope any image that bakes in CAs other than the public ones would provide a way to include others without overriding the expected. Some charts (like gitlab) do an okay job of exposing the option and managing init-containers to copy around the cert files and run `update-ca-certifiaces` (or appropriate tool for the image)..

I think the gap we're trying to think through is mostly the packages that don't have an expected way to manage it (like Grafana). Even then.. it would be nice if all UDS packages "inherited" the trust bundle (for lack of a better term) automagically, similar to how we've abstracted netpols and virtual services to allow us to standardize some defaults across all packages.

**Comment by mjnagel** [comment]
@blancharda yeah that's fair. It's relatively common practice to just document/not document "mount CAs here" as the path to using a custom CA which is not the most pleasant user experience. Even with Gitlab and other charts it tends to be a list of CAs, and if there's a default value...you end up nuking that in an override. 

If we could ease that pain point it would be a value add, I would just hesitate to mutate in 6+ volume mounts and potentially cause other problems esp. if we wiped out default public CAs. Maybe we just add public, common CAs to the trust bundle we build 🤷 and then we can ignore the problem of "wiping out" anything - just would want to validate the need + effect before we add tons of mounts everywhere 😁.

**Comment by blancharda** [comment]
💯 💯 

Same page.
The [draft PR in the other issue](https://github.com/defenseunicorns/uds-bundle-software-factory-nutanix/pull/122/files#diff-91f6da675151956325d901b42e74148629d80bc847c56b42ba425bb0e80aae69R7-R10) (where we're trying this out right now), is adding in the public CAs for that exact reason :) 

Re: tons of mounts.. 
I also agree that doesn't sound ideal..
Still very much just an idea, but if we handled it in the UDS package, maintainers could specify something like:
```
kind: Package
spec:
  trustBundleMounts:
    - targetDir: /etc/pki/ca-trust/extracted/pem/
      format: "pem"
    - targetDir: /etc/pki/ca-trust/extracted/java/cacerts
      format: "jks"
      selector:
        - app.is.garbage: collected
```

So that we could still tailor it to the application/image.. but don't have to dig into the chart to figure out how/where to add the right volume mount settings, or have to worry about where the trust-bundle actually comes from.

Though, 1 pro in favor of the "blast it into all the folders strategy" is that it would be nice to not have to try and figure out what the app actually expected... but idk how realistic that is without putting up with a lot of unnecessary mounts.. Maybe there's a way to be smarter about where to automatically add it 🤔
